src/Umbraco.Web/Editors/MediaController.cs

```diff
@@ -943,5 +944,71 @@ internal static bool CheckPermissions(IDictionary<string, object> storage, IUser
 
             return hasPathAccess;
         }
+
+        /// <summary>
+        /// Returns the references (usages) for the media item
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public MediaReferences GetReferences(int id)
+        {
+            var result = new MediaReferences();
+
+            var relations = Services.RelationService.GetByChildId(id, Constants.Conventions.RelationTypes.RelatedMediaAlias).ToList();
+            var relationEntities = Services.RelationService.GetParentEntitiesFromRelations(relations).ToList();
+
+            var documents = new List<MediaReferences.EntityTypeReferences>();
+            var members = new List<MediaReferences.EntityTypeReferences>();
+            var media = new List<MediaReferences.EntityTypeReferences>();
+
+            foreach (var item in relationEntities)
+            {
+                switch (item)
+                {
+                    case DocumentEntitySlim doc:
```

Comment:
This big switch statement seems to be purely for creating a UDI. It would reduce the code duplication quite a bit if the switch statement simply built a UDI and then a single call to create the `MediaReferences.EntityTypeReferences` since all 3 types are `IContentEntitySlim` which have all of the data needed

Reply:
Yep no problem will fix after standup

Reply:
OK this is for assigning to them the different lists/collection on `MediaReferences`